Add unit tests for D-4 Context Extraction regex patterns

## Background

PR #3889 improved the file path extraction regex patterns to support Python SyntaxError format. While the patterns have been manually tested, we should add unit tests to ensure they work correctly and prevent regressions.

## Scope

Add unit tests for:
- `_extract_file_path_from_error()` - single file extraction
- `_extract_file_paths_from_error()` - multi-file extraction

## Test Cases to Cover

### Pattern 1: Lint format (flake8, pylint, eslint)
- `path/to/file.py:123:45: E501 line too long`
- `src/main.js:10:5: error`

### Pattern 2: Compiler format
- `path/to/file.py(123): error`

### Pattern 3: Error in / File format (with optional quotes)
- `Error in path/to/file.py`
- `File path/to/file.py`
- `File "path/to/file.py"` (quoted)
- `in file src/main.js`

### Pattern 4: Python SyntaxError format
- `File "path/to/file.py", line 10`
- `  File "/usr/lib/python3.10/ast.py", line 50, in parse`
- Multi-line traceback with multiple File references

## Acceptance Criteria

- [ ] Unit tests cover all 4 patterns
- [ ] Tests verify correct extraction (no extra quotes, correct path)
- [ ] Tests verify no false positives
- [ ] Tests added to `orchestrator/tests/` directory

## Related

- PR #3889: feat(d4): improve Context Extraction for Python SyntaxError format
- Issue #3890: D-4 Context Extraction improvements
- Blueprint: EPIC D - Autonomous Coder Agent Family

--- a/handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py
+++ b/handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py
@@ -1174,8 +1174,8 @@ src/file5.py:1:1: error
 src/file6.py:1:1: error
 src/file7.py:1:1: error"""
         result = _extract_file_paths_from_error(error)
-        # Should be limited to 5 files (general_coder_max_files default)
-        assert len(result) <= 5
+        # Should be limited to 5 files (general_coder_max_files in FakeSettings)
+        assert len(result) == 5
 
     def test_extracts_compiler_format(self):
         """Test extraction from compiler format: path/file.py(line): error."""

